src/sentry/incidents/logic.py

```diff
@@ -977,8 +977,8 @@ def update_alert_rule_trigger(
         updated_fields["threshold_type"] = threshold_type.value
     if alert_threshold is not None:
         updated_fields["alert_threshold"] = alert_threshold
-    if resolve_threshold is not None:
-        updated_fields["resolve_threshold"] = resolve_threshold
+    # We set resolve_threshold to None as a 'reset', in case it was previously a value and we're removing it here.
```

Comment:
I'm not sure if this will play well with the thing we do in the serializers where we remove values that haven't changed. If I'm right, then saving a trigger that has a resolve threshold without changing the resolve threshold will set it as None. Could you confirm if that's the case?

Reply:
My gosh, I think you must be right.

I'll fix this.

But I'm not sure how yet. Maybe I won't remove that field.

Reply:
Yeah this looks good then. It might make sense to just require that all the fields are passed all the time, and then remove the  `is not None` checks. Doesn't need to happen in this pr